src/node/stage/approved.rs

```diff
@@ -1022,12 +1021,14 @@ impl Approved {
                 if let Some(index) = self.shared_state.our_history.index_of(&our_key) {
                     if index > dkg_key.1 {
                         trace!("DKG for {:?} already completed", dkg_key);
+                        return;
                     }
                 }
             }
         }
 
         for message in self.dkg_voter.init_dkg_gen(&core.full_id, dkg_key) {
+            self.churn_in_progress = true;
```

Comment:
Should we also set `self.chun_in_progress = false` here, instead of in `promote_and_demote_elders` if `init_dkg_gen` returns empty vec?

Reply:
we cannot do so, as `init_dkg_gen` returns empty could mean there is on-going DKG process, which shall still have `churn_in_progress` as `true` ?